lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1886,7 +1886,25 @@ void TypeChecker::checkIgnoredExpr(Expr *E) {
                     : valueE;
 
     if (auto *Fn = dyn_cast<ApplyExpr>(expr)) {
-      if (auto *calledValue = Fn->getCalledValue()) {
+      /// Some concurrency-related things may have intermediate conversions that
+      /// we want to look through, e.g.
+      ///
+      /// ```swift
+      /// @discardableResult
+      /// @MainActor
+      /// func foo() -> () -> Void {
+      ///   return {}
+      /// }
+      ///
+      /// @MainActor
+      /// func test() {
+      ///   foo() // implicitly converted to `@MainActor @Sendable () -> () ->
+      ///   Void`
+      /// }
+      /// ```
+      /// So we look through such things when trying to locate the called value.
+      if (auto *calledValue =
+              Fn->getCalledValue(/*skipFunctionConversions*/ true)) {
```

Comment:
nitpick on the formatting that we usually do 
```suggestion
              Fn->getCalledValue(/*skipFunctionConversions=*/true)) {
```